content/docs/gke/authentication.md

```diff
@@ -7,15 +7,89 @@ weight = 4
 [provision three service accounts](https://www.kubeflow.org/docs/gke/deploy/deploy-cli/#gcp-service-accounts) with different
-privileges. In particular, the `${KFAPP}-user` service account is meant to grant your user services access to GCP. 
+privileges. In particular, the `${KFAPP}-user` service account is meant to grant your user services access to GCP.
 The credentials to this service account can be accessed within the cluster as a
 [Kubernetes secret](https://kubernetes.io/docs/concepts/configuration/secret/).
 
 The secret will have basic access to a limited set of GCP services by default, but more roles can be granted through the
 [GCP IAM console](https://console.cloud.google.com/iam-admin/).
 
+### Namespaces
+
+Starting from Kubeflow version 0.6, you can consume Kubeflow in individual namespaces. The `kubeflow` namespace is
+intended for running Kubeflow system components while individual jobs and model deployments are expected to run in
+separate namespaces. In order to do this you will need to install GCP credentials into the new namespace and create
+a PodDefault object to attach the credentials to certain pods.
+
+##### Credentials
+
+You can add credentials to the new namespace by either copying them from an existing Kubeflow namespace or by
+creating a new service account.
+
+To copy credentials from one namespace to another namespace use the following CLI commands (**Note**: there is an
+[issue](https://github.com/kubeflow/kubeflow/issues/3640) filed to automate these commands):
+
+```
+NAMESPACE=<new kubeflow namespace>
+SOURCE=kubeflow
+NAME=user-gcp-sa
+SECRET=$(kubectl -n ${SOURCE} get secrets ${NAME} -o jsonpath="{.data.${NAME}\.json}" | base64 -d)
+kubectl create -n ${NAMESPACE} secret generic ${NAME} --from-literal="${NAME}.json=${SECRET}"
+```
+
+To create a new service account instead of copying credentials, use the following steps:
+
+1. Create a service account with the desired roles
+```
+export PROJECT_ID=<GCP project id>
+export NAMESPACE=<new kubeflow namespace>
+export SA_NAME=<service account name>
+export GCPROLES=roles/editor
+gcloud --project=${PROJECT_ID} iam service-accounts create $SA_NAME
+gcloud projects add-iam-policy-binding $PROJECT_ID \
+    --member serviceAccount:$SA_NAME@$PROJECT_ID.iam.gserviceaccount.com \
+    --role $GCPROLES
+```
+
+2. Download the JSON service account key, set `KEYPATH` to the correct path, and create the key 
+```
+export KEYPATH=some/path/${SA_NAME}.gcp.json
+gcloud --project=${PROJECT_ID} iam service-accounts keys create ${KEYPATH} \
+   --iam-account $SA_NAME@$PROJECT_ID.iam.gserviceaccount.com
+```
+
+3. Upload the JSON service account key to cluster as a secret
+```
+kubectl create secret generic user-gcp-sa -n $NAMESPACE \ --from-file=user-gcp-sa.json=${KEYPATH}
+```
+
+##### PodDefault Object
+
+Create a pod default in a file called `add-gcp-secret.yaml` and apply it using: `kubectl apply -f add-gcp-secret.yaml -n $NAMESPACE`
```

Comment:
Can you explain the PodDefault object a bit here? Why it's needed/what it does

Reply:
Good idea:

I think the key ideas are

1. PodDefault will match all pods matched by the selector
1. PodDefault will modify the pod to inject the volume, secrets and environment variable listed in the PodDefault

This is a way to centrally manage config that should be added to all pods.

Reply:
Done.